sdk/src/main/scala/com/horizen/SidechainState.scala

```diff
@@ -376,7 +376,8 @@ class SidechainState private[horizen] (stateStorage: SidechainStateStorage,
       val keyRotationProof = keyRotationTransaction.getKeyRotationProof
       val oldCertifiersKeys = certifiersKeys(withdrawalEpoch - 1).get
 
-      val messageToSign = keyRotationProof.newKey.getHash
+      val messageToSign = CryptoLibProvider.thresholdSignatureCircuitWithKeyRotation
+        .getMsgToSignForSigningKeyUpdate(keyRotationProof.newKey.pubKeyBytes(), withdrawalEpoch, params.sidechainId)
```

Comment:
here we should match `keyRotationProof.keyType`
case signing -> getMsgToSignForSigningKeyUpdate
case master -> getMsgToSignForMasterKeyUpdate

Reply:
Done